### PR TITLE
Make `ReadonlyBasicProperties` a class.

### PR DESCRIPTION
**Pull request by lukebakken**
Follow-up to #1601

https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1601#issuecomment-2171885716

cc @bollhals

### CODE REVIEW

File: projects/RabbitMQ.Client/client/api/ReadonlyBasicProperties.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ namespace RabbitMQ.Client
     /// <summary>
     /// AMQP specification content header properties for content class "basic"
     /// </summary>
-    public readonly struct ReadOnlyBasicProperties : IReadOnlyBasicProperties
+    public class ReadOnlyBasicProperties : IReadOnlyBasicProperties
```
Comment (Tornhoof) [inline projects/RabbitMQ.Client/client/api/ReadonlyBasicProperties.cs:42]: seal it?

Comment (lukebakken) [reply projects/RabbitMQ.Client/client/api/ReadonlyBasicProperties.cs:42]: Thank you for noticing! @Tornhoof - always feel free to use the "Review" function in GitHub on a PR here. It should allow anyone to provide feedback.

**Comment by bollhals** [comment]
Sorry a bit too late to the party :)

Doesn't that leave BasicProperties being a struct? Doesn't make much sense to me to convert one but not the other.

**Comment by lukebakken** [comment]
@bollhals 💥 https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1609

**Pull request merged**